Bottom: be7517860ae66e6bdbc1ba38f69ed1821a6c6820
Top:    80e65477b7f05bc122b8b66c68b035935e4379b4
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-17 19:15:38 +0800

Refresh of virtio-net-small-buffer-xdp

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 6d1b1ab..d12e254 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -399,7 +399,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 	unsigned int headroom = vi->hdr_len + VIRTNET_RX_PAD + xdp_headroom;
 	unsigned int buflen = SKB_DATA_ALIGN(GOOD_PACKET_LEN + headroom) +
 		              SKB_DATA_ALIGN(sizeof(struct skb_shared_info));
-	unsinged int delta = 0;
+	unsigned int delta = 0;
 	len -= vi->hdr_len;
 
 	rcu_read_lock();
@@ -457,7 +457,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 err_xdp:
 	rcu_read_unlock();
 	dev->stats.rx_dropped++;
-	kfree_skb(skb);
+	put_page(virt_to_head_page(buf));
 xdp_xmit:
 	return NULL;
 }
